atd-toolbox/database_load_test/database_writer.py

```diff
@@ -0,0 +1,335 @@
+            placeholders.append("%s")
+
+        elif field["data_type"] == "jsonb":
+            fields.append(field["column_name"])
+            values.append(random_jsonb(field))
+            placeholders.append("%s")
+
+        else:
+            print("We don't know how to handle this type: ", field["data_type"])
+            quit()
+
+    insert_sql = f"insert into {table} ({', '.join(fields)}) values ({', '.join(placeholders)}) returning *"
+    return (insert_sql, values)
+
+
+def random_character_varying(field):
+    fake = Faker()
+    fake.add_provider(lorem)
+    string = fake.paragraph()[: field["character_maximum_length"]]
+    return string
+
+
+def random_integer(field):
+    precision = field["numeric_precision"] - 1
+    integer = random.randint(0, 2**precision)
+    return integer
+
+
+def random_date(field):
+    fake = Faker()
+    fake.add_provider(date_time)
+    date = fake.date_this_year()
+    return date
+
+
+def random_timestamp_with_time_zone(field):
+    fake = Faker()
+    fake.add_provider(date_time)
+    timestamp = fake.date_time_this_year()
+    return timestamp
+
+
+def random_timestamp_without_time_zone(field):
+    return random_timestamp_with_time_zone(field)
+
+
+def random_time_with_time_zone(field):
+    fake = Faker()
+    fake.add_provider(date_time)
+    time = fake.time()
+
+
+def random_time_without_time_zone(field):
+    return random_time_with_time_zone(field)
+
+
+def random_double_precision(field):
+    fake = Faker()
+    fake.add_provider(python)
+    float = fake.pyfloat()
+
+
+def random_text(field):
+    fake = Faker()
+    fake.add_provider(lorem)
+    string = "\n".join(fake.paragraphs(nb=5))
+    return string
+
+
+def random_boolean(field):
+    fake = Faker()
+    fake.add_provider(python)
+    boolean = fake.pybool()
+    return boolean
+
+
+def random_numeric(field):
+    if not field["numeric_scale"]:
+        field["numeric_scale"] = 2
+    if not field["numeric_precision"]:
+        field["numeric_precision"] = 10
+    precision = field["numeric_precision"] - field["numeric_scale"]
+    fake = Faker()
+    fake.add_provider(python)
+    # these names feel backwards, but this is what it is
+    numeric = fake.pyfloat(left_digits=precision, right_digits=field["numeric_scale"])
+    return numeric
+
+
+def random_json(field):
+    fake = Faker()
+    fake.add_provider(misc)
+    json = fake.json()
+    return json
+
+
+def random_jsonb(field):
+    return random_json(field)
+
+
+def random_position():
+    latitude = 30.274722 + random.uniform(-0.1, 0.1)
+    longitude = -97.740556 + random.uniform(-0.1, 0.1)
+    position = f"ST_GeomFromEWKT('SRID=4326;POINT({longitude} {latitude})')"
+    return position
+    pass
+
+
+def get_next_crash_id():
+    primary = get_primary_connection()
+    cursor = primary.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+    cursor.execute("SELECT max(crash_id)+1 as next_id FROM atd_txdot_crashes;")
```

Comment:
Because we get `crash_id`s from CRIS, we don't actually maintain a sequence for the crash table, and this is the only way to know you're not going to try to reuse a value in your insert.